common/lib/xmodule/xmodule/video_module/transcripts_utils.py

```diff
@@ -468,6 +470,37 @@ def get_or_create_sjson(item, transcripts):
     return sjson_transcript
 
 
+def get_video_transcript_data(video, lang_code):
+    """
+    Gets video transcript(s) from edx-val for either an edx_video_id or any of the external sources.
+
+    Arguments:
+        video(VideoDescriptor): A video XModule descriptor.
+        lang_code(str): Language code
+    """
+    video_transcript_data = None
+    if video.edx_video_id:
+        # Check transcripts for an `edx_video_id`
+        video_transcript_data = edxval_api.get_video_transcript(video_id=video.edx_video_id, language_code=lang_code)
+        # TODO: Make decision about actual external sources if we don't get transcript for edx_video_id.
+    else:
+        # Check if transcripts are there for any of the external source.
+        external_video_ids = [video.youtube_id_1_0] + get_html5_ids(video.html5_sources)
```

Comment:
@Qubad786 Can we re-write this function like

```python
def get_video_transcript_data(video, lang_code):
    """
    Gets video transcript(s) from edx-val for either an edx_video_id or any of the external sources.

    Arguments:
        video(VideoDescriptor): A video XModule descriptor.
        lang_code(str): Language code
    """
    video_transcript_data = None
    video_ids = [video.edx_video_id] + [video.youtube_id_1_0] + get_html5_ids(video.html5_sources)

    for video_id in video_ids:
        if not video_id:
            continue

        video_transcript_data = edxval_api.get_video_transcript(
            video_id=video_id,
            language_code=lang_code
        )
        if video_transcript_data:
            break
        
    return video_transcript_data
```

Any concerns?

Reply:
There is one thing that we may add in future, that's solely depend on `edx_video_id`. I've added `TODO `for that too. That's the only concern with this approach.